library/std/tests/env.rs

```diff
@@ -1,12 +1,24 @@
 use std::env::*;
 use std::ffi::{OsStr, OsString};
 
-use rand::distributions::Alphanumeric;
-use rand::{thread_rng, Rng};
+use rand::distributions::{Alphanumeric, DistString};
+
+/// Copied from `std::test_helpers::test_rng`, since these tests rely on the
+/// seed not being the same for every RNG invocation too.
+#[track_caller]
+pub(crate) fn test_rng() -> rand_xorshift::XorShiftRng {
+    use core::hash::{BuildHasher, Hash, Hasher};
+    let mut hasher = std::collections::hash_map::RandomState::new().build_hasher();
```

Comment:
Wait, this means we already pull in non-determinism on platforms that have have a randomness source, right? Then my previous concern is moot for std tests.

But alloc tests still need a similar solution.

Reply:
I'll just use this there. It's gross but I'm doing the `cfg` approach now and it's also ending up really gross, and wouldn't remove this grossness.

Reply:
I've removed all use of hard-coded seeds from the tests (we still use them in benches, which is intentional, since nondeterminism is bad in those).

This was kind of a pain in the ass, but it would have been painful anyway I guess.